Remotion/Web/Development.WebTesting.IntegrationTests/FileDownloadTest.cs

```diff
@@ -174,17 +175,19 @@ public void TestDownloadOpensInNewWindow_DownloadTimeoutExceeded_WithUnknownFile
       {
         startDownloadLambda();
 
-        Assert.That (
-          () => Helper.BrowserConfiguration.DownloadHelper.HandleDownloadWithDetectedFileName (downloadStartedTimout, downloadUpdatedTimeout),
-          Throws.InstanceOf<DownloadResultNotFoundException>()
-              .With.Message.StartsWith (
-                string.Format (
-                  "The download result file did not get updated for longer than the downloadUpdatedTimeout of '{0}'. The download appears to have failed.",
-                  downloadUpdatedTimeout)));
-
+        // Kill the browser session after the test action completes to avoid waiting longer than necessary
+        // and spin it up again for subsequent tests. Make sure to await it at the end of the test.
+        var reinitializeFixtureTask = ReinitializeFixtureAfterDelayAsync (downloadUpdatedTimeout + TimeSpan.FromSeconds (3));
 
-        //The browser continues writing in the download file, therefore we have to wait a little bit so that it does not interfere with the next test.
-        Thread.Sleep (TimeSpan.FromSeconds (3));
+        Assert.That (
+            () => Helper.BrowserConfiguration.DownloadHelper.HandleDownloadWithDetectedFileName (downloadStartedTimout, downloadUpdatedTimeout),
+            Throws.InstanceOf<DownloadResultNotFoundException>()
+                .With.Message.StartsWith (
+                    string.Format (
+                        "The download result file did not get updated for longer than the downloadUpdatedTimeout of '{0}'. The download appears to have failed.",
+                        downloadUpdatedTimeout)));
+
+        reinitializeFixtureTask.Wait();

```

Comment:
Task.Wait will rethrow the excpetions from inside the run-method wrapped in an aggregate-exception with a not-helpful message.